oremda/plasma_client.py

```diff
@@ -16,40 +16,28 @@ def get_object(self, object_id: plasma.ObjectID) -> np.ndarray:
         return self.plasma_client.get(object_id)
 
 
-class PlasmaArray(BaseModel):
-
-    client: PlasmaClient
-    object_id: plasma.ObjectID
-
-    class Config:
-        arbitrary_types_allowed = True
-
-    def __init__(self, client, object_id, **data):
-        super().__init__(client=client, object_id=object_id, **data)
-
-    @validator("object_id", pre=True)
-    def validate_object_id(cls, id, values):
-        client = values["client"]
-        conversions = {
-            plasma.ObjectID: lambda x: x,
-            str: lambda x: plasma.ObjectID(bytes.fromhex(x)),
-            np.ndarray: lambda x: client.create_object(x),
-        }
-
-        id_type = type(id)
-        if id_type not in conversions:
-            raise TypeError(f"Cannot convert type {id_type} to ObjectID")
-
-        return conversions[id_type](id)
+class PlasmaArray(DataArray):
+    def __init__(self, client: PlasmaClient, id_or_data: Union[str, plasma.ObjectID, DataType]):
+        self.client = client
+        self.object_id: ObjectId
+        if isinstance(id_or_data, plasma.ObjectID):
+            object_id: Any = id_or_data
+            self.object_id = object_id
```

Comment:
We do to avoid typing errors.
Normally, if a variable is of a union type (e.g. `var: TypeA | TypeB`), we can use an `if isinstance(var, TypeA)` block to narrow its type down to `TypeA`.
However, the class `plasma.ObjectID` is really interpreted as being type `Unknown` (due to reasons that escape me, but I think it has to do with the fact that the implementation is written in C), so we need to do have some workaround to stop the type system to complain.
This is also the same reason why I introduced the `ObjectId` class, which is just an interface for `plasma.ObjectID`.